sbin/functions.sh

```diff
@@ -982,6 +982,8 @@ installer_main()
 	local p1
 	local p2
 	# XXX: TODO. the partition name should be returned by create_partition
+	while ([ -z ${p1} ] || [ -z ${p2} ])
```

Comment:
This can now potentially be an infinite loop.

What's causing the delay ? The propagation of the partitions to devfs ? Something else ?

Either way, we should document the reason we need to loop, and have an exit condition if they never do appear (and then exit the process with an appropriate error message).